### PR TITLE
Adds gitlab based CI/CD

### PR DESCRIPTION
**Pull request by jovial**

### CODE REVIEW

**Review by markgoddard — COMMENTED** [review]
Nice to see this all come together.

File: .automation.conf/tempest/tempest-preprod.overrides.conf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+http_timeout = 600
+[service_available]
+cinder = false
+
+[compute-feature-enabled]
+resize = true
+live_migration = true
+
+[network]
+public_network_id = 13047566-bbf7-4e1d-941b-58e7b571ca8d
```
Comment (JohnGarbutt) [inline .automation.conf/tempest/tempest-preprod.overrides.conf:28]: I guess we need to some change this between prod and preprod, eventually.

**Review by JohnGarbutt — APPROVED** [review]
Nice, real progress here.

File: .automation.conf/tempest/tempest-preprod.overrides.conf (hunk 1 above)
Comment (jovial) [reply .automation.conf/tempest/tempest-preprod.overrides.conf:28]: yeah, the plan was to use the ENVIRONMENT environment variable to switch to using a different file

**Pull request merged**